src/Test/Html/Selector.elm

```diff
@@ -309,6 +354,54 @@ checked =
 -- HELPERS
 
 
+findAttributeInFacts : Bool -> String -> ElmHtml.InternalTypes.Facts a -> Selector
+findAttributeInFacts caseInsensitive name facts =
+    let
+        lookup =
+            case caseInsensitive of
+                True ->
+                    dictGetCaseInsensitive
+
+                False ->
+                    Dict.get
+    in
+    facts.stringAttributes
+        |> lookup name
+        |> Maybe.map (namedAttr name)
+        |> orElseLazy
+            (\_ ->
+                facts.boolAttributes
+                    |> lookup name
+                    |> Maybe.map (namedBoolAttr name)
+            )
+        |> Maybe.withDefault Invalid
+
+
+findInList : (a -> Bool) -> List a -> Maybe a
+findInList predicate list =
+    case list of
+        [] ->
+            Nothing
+
+        first :: rest ->
+            if predicate first then
+                Just first
+            else
+                findInList predicate rest
+
+
+dictGetCaseInsensitive : String -> Dict String a -> Maybe a
+dictGetCaseInsensitive key dict =
+    let
+        keyToLower =
+            String.toLower key
+    in
+    dict
+        |> Dict.toList
+        |> findInList (\( k, v ) -> String.toLower k == keyToLower)
```

Comment:
So initially we were trying to ensure that our lookup of attributes vs properties matched the semantics of the actual DOM, since it doesn't care about the casing of attribute keys but does care about properties. Looking at it again, though, I don't think we actually need to do this since it isn't `virtual-dom` that is making this conversion but the actual DOM during patching. We would just want to make sure that we catch if `String.toLower name` is equal to `"class"`